src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -341,88 +421,73 @@ define([
-      });
+      return (
+        <AppModalComponent
+          activeTask={this.state.activeTask}
+          appVersionsFetchState={this.state.appVersionsFetchState}
+          destroyApp={this.destroyApp}
+          fetchTasks={this.fetchTasks}
+          fetchAppVersions={this.fetchAppVersions}
+          model={this.state.activeApp}
+          onDestroy={this.modalDestroy}
+          onShowTaskDetails={this.handleShowTaskDetails}
+          onShowTaskList={this.handleShowTaskList}
+          onTasksKilled={this.handleTasksKilled}
+          rollBackApp={this.rollbackToAppVersion}
+          scaleApp={this.scaleApp}
+          suspendApp={this.suspendApp}
+          tasksFetchState={this.state.tasksFetchState}
+          router={this.props.router}
+          ref="modal" />
+      );
     },
 
-    onTabClick: function(id) {
-      this.setState({
-        activeTabId: id
-      });
+    routeDeployments: function() {
+      return null;
+    },
 
-      if (id === tabs[0].id) {
-        this.setPollResource(this.fetchApps);
-      } else if (id === tabs[1].id) {
-        this.setPollResource(this.fetchDeployments);
-      }
+    routeNewapp: function() {
+      return (
+        <NewAppModalComponent
+          model={this.state.activeApp}
+          onDestroy={this.modalDestroy}
+          onCreate={this.handleAppCreate}
+          ref="modal" />
+      );
     },
 
     render: function() {
       var modal;
+      var route = this.state.route;
 
-      /* jshint trailing:false, quotmark:false, newcap:false */
-      if (this.state.modalClass === AppModalComponent) {
-        modal = (
-          <AppModalComponent
-            activeTask={this.state.activeTask}
-            appVersionsFetchState={this.state.appVersionsFetchState}
-            destroyApp={this.destroyApp}
-            fetchTasks={this.fetchTasks}
-            fetchAppVersions={this.fetchAppVersions}
-            model={this.state.activeApp}
-            onDestroy={this.handleModalDestroy}
-            onShowTaskDetails={this.handleShowTaskDetails}
-            onShowTaskList={this.handleShowTaskList}
-            onTasksKilled={this.handleTasksKilled}
-            rollBackApp={this.rollbackToAppVersion}
-            scaleApp={this.scaleApp}
-            suspendApp={this.suspendApp}
-            tasksFetchState={this.state.tasksFetchState}
-            ref="modal" />
-        );
-      } else if (this.state.modalClass === NewAppModalComponent) {
-        modal = (
-          <NewAppModalComponent
-            model={this.state.activeApp}
-            onDestroy={this.handleModalDestroy}
-            onCreate={this.handleAppCreate}
-            ref="modal" />
-        );
-      } else if (this.state.modalClass === AboutModalComponent) {
-        modal = (
-          <AboutModalComponent
-            onDestroy={this.handleModalDestroy}
-            ref="modal" />
-        );
+      if(route) {
+        var routeName = _.str.capitalize(route.name);
```

Comment:
[`_.str.capitalize`](https://github.com/epeli/underscore.string/blob/master/capitalize.js) is a one-line function. Since you're only using it in one place, you could cut out this entire library:

```
route.name.charAt(0).toUpperCase() + route.name.slice(1)
```